crypto/types.go

```diff
@@ -110,20 +111,22 @@ type Signature []byte
 // It allows a function caller differentiate unexpected program errors from errors caused by
 // invalid inputs.
 type InvalidInputsError struct {
```

Comment:
I think the constructor is private because we want these errors to only be produced from within this package. In that case this struct should be private as well to avoid direct instantiation.

Reply:
I also prefer to have it private. I thought having it public allows callers to test for the error type without using `IsInvalidInputsError`